diff --git a/src/gallium/drivers/nouveau/nvc0/nvc0_program.c b/src/gallium/drivers/nouveau/nvc0/nvc0_program.c
index 8851503..691b9a9 100644
--- a/src/gallium/drivers/nouveau/nvc0/nvc0_program.c
+++ b/src/gallium/drivers/nouveau/nvc0/nvc0_program.c
@@ -593,6 +593,9 @@ nvc0_program_translate(struct nvc0_program *prog, uint16_t chipset,
    info->target = chipset;
 
    info->bin.nir = nir_shader_clone(NULL, prog->nir);
+   if (info->bin.nir) {
+      nir_lower_robust_access(info->bin.nir, NULL, NULL);
+   }
 
 #ifndef NDEBUG
    info->target = debug_get_num_option("NV50_PROG_CHIPSET", chipset);
